compose_area: Fix compose_area cut pasted message bigger than max_length.

Signior-X opened the pull request:
Follow up for https://github.com/zulip/zulip/pull/16061

I have changed the way we were going to solve a bit. The highlighter feature was using a wrapper and I don't think it is good way to fake the highlight. Instead what this does it, it allows writing longer than MAX_LIMIT but also show the error above until, the length is more than the limit and removes it as it gets back in limit.
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
Fixes #15909

**Testing plan:** <!-- How have you tested? -->
In progress. I will add tests soon. Tested with node tests. Once finalized the behavior, will add node tests.




<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->
[CZO link here](https://chat.zulip.org/#narrow/stream/6-frontend/topic/compose.20long.20message.20.2318871)

timabbott commented on static/js/compose.js:
```diff
@@ -45,6 +45,8 @@ import * as upload from "./upload";
 import * as util from "./util";
 import * as zcommand from "./zcommand";
 
+const MAX_MESSAGE_LENGTH = 10000;
```
Sure; feel free to coordinate with @ligmitz who's moved a bunch of similar things in recent weeks.

timabbott commented:
Ideally, we'd have a nicer notice that showed the number of characters they have and the maximum.  A lot other tools that have a max length limit do this nicely.

Signior-X commented:
@timabbott I have updated the PR and now it shows an indicator.

![image](https://user-images.githubusercontent.com/56730716/124383795-03424780-dcec-11eb-84bb-3cfd035beba2.png)

![image](https://user-images.githubusercontent.com/56730716/124383798-09382880-dcec-11eb-89ac-7b708dd54e92.png)

Signior-X commented:
@timabbott this is now ready. Rebased this.
Also fixed the alignment a bit also.
![image](https://user-images.githubusercontent.com/56730716/124744270-367b1580-df3c-11eb-9f15-ed28fff8d101.png)

timabbott commented on static/templates/compose.hbs:
```diff
@@ -100,6 +100,7 @@
                                         <span></span>{{t 'Press Enter to send' }}
                                     </label>
                                 </div>
+                                <p id="compose_limit_indicator"></p>
```
This is more naturally a span; changed that detail.

timabbott commented on static/styles/compose.css:
```diff
@@ -303,6 +317,10 @@ textarea.new_message_textarea {
     margin-top: 5px;
+
+    &.over_limit {
+        border: 1px solid hsl(0, 100%, 50%) !important;
```
I changed this to a box-shadow, which fixes the artifacts on the right corners.

timabbott commented on static/js/compose_validate.js:
```diff
@@ -491,6 +514,21 @@ export function validate() {
+                },
+            ),
+            $("#compose-textarea"),
+        );
```
I think we don't need this error message; the red box and red warning tell you what you did wrong, and it's kinda ugly :).  So I just removed the error message part and replaced it with a comment explaining that reasoning.

timabbott commented:
This is great, merged with the small changes noted above, thanks @Signior-X!
